File: Client/src/Message.cpp
//
// Created by yadinben@wincs.cs.bgu.ac.il on 10/01/2020.
//

#include "Message.h"
Message::Message() : subscription(), MessageId(), destination(), frameBody() {}

void Message::setSubscription(const string &subscription) {
    Message::subscription = subscription;
}

void Message::setMessageId(const string &messageId) {
    MessageId = messageId;
}

const string &Message::getDestination() const {
    return destination;
}

void Message::setDestination(const string &destination) {
    Message::destination = destination;
}

const string &Message::getFrameBody() const {
    return frameBody;
}

void Message::setFrameBody(const string &frameBody) {
    Message::frameBody = frameBody;
}

File: Server/src/main/java/bgu/spl/net/impl/BookClubService/Messages/MESSAGE.java
package bgu.spl.net.impl.BookClubService.Messages;

import bgu.spl.net.impl.BookClubService.Frame;
import bgu.spl.net.impl.BookClubService.StompFrame;

import java.util.HashMap;
import java.util.Map;

public class MESSAGE implements Frame {
    private StompFrame frame;
    private String subId;
    private int msgId;
    private String genre;

    public MESSAGE(StompFrame frame , String subId , int msgId) {
        this.frame = frame;
        this.genre = frame.getHeader().get("destination");
        this.subId = subId;
        this.msgId = msgId;

    }

    @Override
    public int getOperationCode() {
        return -3;
    }


    @Override
    public StompFrame getBuiltFrame() {
        StompFrame sendFrame = new StompFrame();
        sendFrame.setStompCommand("MESSAGE");
        Map<String , String> headers = new HashMap<>();
        headers.put("subscription" , subId+"");
        headers.put("Message-id" , msgId+"");
        headers.put("destination" , genre);
        sendFrame.setHeader(headers);
        sendFrame.setFrameBody(frame.getFrameBody());

        return sendFrame;
    }
}
